fem/fe.cpp

```diff
@@ -8161,7 +8161,7 @@ L2_SegmentElement::L2_SegmentElement(const int p, const int type)
 
 #ifndef MFEM_THREAD_SAFE
    shape_x.SetSize(p + 1);
-   dshape_x.SetDataAndSize(NULL, p + 1);
+   dshape_x.SetDataAndSize((double*)NULL, p + 1);
```

Comment:
Is the `double*` cast necessary here?

Reply:
Not necessarily, but we would sacrifice the const fixes that go along with this change (I suppose it could be made a separate pull request). In order to fix those I had to add const versions of certain methods, and when adding those the Koenig lookup doesn't know which one to use.

Here are two (of many more) cases of const issues:

  * https://github.com/mfem/mfem/pull/284/files#diff-c291e54b7a540114c9acb5b451b4c2aeL367
  * https://github.com/mfem/mfem/pull/284/files#diff-05d857784730cd0e013ff5f05d401491L2048

Both cases didn't turn out to be a problem, but that is just because everyone has been careful so far. The usage is unsafe in general and it could make it difficult to track down bugs if the state of const objects are unknowingly changed.